clarity/contracts/arkadiko-multi-hop-swap-v1-1.clar

```diff
@@ -0,0 +1,37 @@
+;; @contract Multihop Swap - Decentralised exchange
+;; @version 1
+
+(use-trait ft-trait .sip-010-trait-ft-standard.sip-010-trait)
+
+(define-constant ERR-TOO-MUCH-SLIPPAGE u7771)
+(define-constant ERR-SWAP-ONE-FAILED u7772)
+(define-constant ERR-SWAP-TWO-FAILED u7773)
+
+;; Swap X for Z with token Y in between
+;; For instance, I want to swap STX to xBTC but there is no STX/xBTC pair
+;; I swap STX to USDA and USDA to xBTC since there is a xBTC/USDA pair
+(define-public (swap-x-for-z
+  (token-x-trait <ft-trait>)
+  (token-y-trait <ft-trait>)
+  (token-z-trait <ft-trait>)
+  (dx uint)
+  (min-dz uint)
+  (inverse-first bool)
+  (inverse-second bool)
+)
+  (let (
+    (swapped-amounts (if inverse-first
+      (unwrap! (contract-call? .arkadiko-swap-v2-1 swap-y-for-x token-x-trait token-y-trait dx u0) (err ERR-SWAP-ONE-FAILED))
+      (unwrap! (contract-call? .arkadiko-swap-v2-1 swap-x-for-y token-x-trait token-y-trait dx u0) (err ERR-SWAP-ONE-FAILED))
+    ))
+    (y-amount (unwrap-panic (element-at swapped-amounts u1)))
```

Comment:
Q: The function `swap-y-for-x` returns `(list dx dy)` where `dy` is the input and `dx` is the result. So we should get the first element from the list instead of the second?

Reply:
Yes that's a good point, I also need to do it conditional here. Will also write a test